src/main/java/de/tum/in/www1/artemis/service/AuthorizationCheckService.java

```diff
@@ -333,4 +337,18 @@ public boolean isUserAllowedToGetResult(Exercise exercise, StudentParticipation
                 && (exercise.getAssessmentDueDate() == null || exercise.getAssessmentDueDate().isBefore(ZonedDateTime.now())) && result.getAssessor() != null
                 && result.getCompletionDate() != null;
     }
+
+    /**
+     * Checks if the user is only student in the course.
+     *
+     * If the exam is over and the results are not published, he is not allowed to request a result.
```

Comment:
I would improve the documentation here as it might be hard to understand what the separate conditions actually check (first see my next comment).
```suggestion
     * Checks if the user is allowed to see the exam result. Returns true if
     * - the current user is at least teaching assistant in the course
     * - OR if the exam has already ended and the results were published
```

Reply:
The idea was that a student is also allowed to see it if the the exam has not ended yet. I've adjusted that in the documentation